src/main/java/org/broadinstitute/hellbender/tools/funcotator/Funcotator.java

```diff
@@ -482,34 +486,25 @@ private void enqueueAndHandleVariant(final VariantContext variant, final Referen
             }
         }
 
-        // Get our VariantContext for annotation:
-        final VariantContext variantContextFixedContigForDataSources = variantContextBuilderForFixedContigForDataSources.make();
-
         // Create a place to keep our funcotations:
         final List<Funcotation> funcotations = new ArrayList<>();
 
         // Annotate with Gencode first:
         // Create a list of GencodeFuncotation to use for other Data Sources:
         final List<GencodeFuncotation> gencodeFuncotations = new ArrayList<>();
 
-        for ( final GencodeFuncotationFactory factory : gencodeFuncotationFactories ) {
-            final List<Funcotation> funcotationsFromGencodeFactory = factory.createFuncotations(variantContextFixedContigForDataSources, referenceContext, featureSourceMap);
-            funcotations.addAll(funcotationsFromGencodeFactory);
-            gencodeFuncotations.addAll(
-                    funcotationsFromGencodeFactory.stream()
-                    .map(x -> (GencodeFuncotation)x)
-                    .collect(Collectors.toList())
-            );
-        }
-
-        // Annotate with the rest of the data sources:
-        for ( final DataSourceFuncotationFactory funcotationFactory : dataSourceFactories ) {
-
-            // Make sure we don't double up on the Gencodes:
-            if ( funcotationFactory.getType().equals(FuncotatorArgumentDefinitions.DataSourceType.GENCODE) ) {
-                continue;
+        // Perform the actual annotation.  Note that we leverage the ordering of datasources here.
```

Comment:
Please add another comment here explicitly calling out that the comparator guarantees that gencode data sources are all first in the list.

Reply:
Done.